src/main/java/org/jabref/gui/actions/ActionHelper.java

```diff
@@ -81,6 +81,39 @@ public static BooleanExpression isFilePresentForSelectedEntry(StateManager state
         return BooleanExpression.booleanExpression(fileIsPresent);
     }
 
+    /**
+     * Check if at least one of the selected entries has linked files
+     * <br>
+     * Used in {@link org.jabref.gui.maintable.OpenExternalFileAction} when multiple entries selected
+     * @param stateManager manager for the state of the GUI
+     * @param preferencesService services of preferences
+     * @return a boolean binding
+     */
+    public static BooleanExpression hasPresentFileForSelectedEntries(StateManager stateManager, PreferencesService preferencesService) {
+        return Bindings.createBooleanBinding(() -> {
+            ObservableList<BibEntry> selectedEntries = stateManager.getSelectedEntries();
+            List<LinkedFile> files;
+            for (BibEntry entry:selectedEntries) {
+                files = entry.getFiles();
+
+                if ((entry.getFiles().size() > 0) && stateManager.getActiveDatabase().isPresent()) {
+                    if (files.get(0).isOnlineLink()) {
+                        return true;
+                    }
+
+                    Optional<Path> filename = FileHelper.find(
```

Comment:
No need to check the filename for presence here, just later on try to open the file. Either it works or not.